src/lib/peer.ts

```diff
@@ -1,24 +1,38 @@
 import { LambdasClient } from 'dcl-catalyst-client'
 import { Profile } from '../modules/profile/types'
 import { BaseAPI } from './api'
+import { ProfileEntity } from './types'
 
 export class PeerAPI extends BaseAPI {
-  cache: Record<string, Promise<Profile>> = {}
+  cache: Record<string, Profile> = {}
   lambdasClient: LambdasClient
 
   constructor(url: string) {
     super(url)
     this.lambdasClient = new LambdasClient(`${url}/lambdas`)
   }
 
-  fetchProfile = async (address: string) => {
+  // TODO, how many ways of retrieving a profile are out there?
+  public async fetchProfile(address: string): Promise<Profile> {
     if (address in this.cache) {
       return this.cache[address]
     }
-    const promise = this.lambdasClient
-      .fetchProfiles([address.toLowerCase()])
-      .then(profiles => profiles[0])
-    this.cache[address] = promise
-    return promise
+    const profiles = await this.lambdasClient.fetchProfiles([
+      address.toLowerCase()
+    ])
+    this.cache[address] = profiles[0]
```

Comment:
This is caching _after_ the response has come back from the catalyst. The way it was done before (storing the promise in the cache) was so if two requests for the same profile happen simultaneously, only one request would go to the server.

This is very common when using the `<Profile />` container ([link](https://github.com/decentraland/decentraland-dapps/tree/master/src/containers/Profile)), pages like this one would send several requests for the same profile:

<img width="1201" alt="Screen Shot 2021-07-27 at 11 57 46 AM" src="https://user-images.githubusercontent.com/2781777/127177496-0333ee5e-02c8-4aee-a07e-813fa05364d7.png">

Reply:
Ohhh, great catch! I didn't know that it was built that way to prevent multiple requests from happening.